<!-- Launch file for the Depth-Based-RealTime-RRTSTAR-detector -->

<launch>

<arg name="my_stack_directory" 		default="$(env AEROSTACK_STACK)" />
<arg name="drone_id_int" 		      default="7" />  	
<arg name="drone_id_namespace"    default="drone7" />
<arg name="configs_path"			default="$(find Depth_BASED_RRT_STAR)/cfg/DepthBasedNavigation.xml" />
<arg name="Xmin" 						default="-4.0"/><!-- x:-4-4, y:-2.2-2.2 -->
<arg name="Xmax" 						default="4.0"/>
<arg name="Ymin" 						default="-4"/>
<arg name="Ymax" 						default="4"/>
<arg name="END_X" 						default="3.0"/><!-- x:3, y:0 -->
<arg name="END_Y" 						default="0.0"/>
<arg name="END_Z" 						default="1.1"/>
<arg name="RRTSTAR_NEIGHBOR_FACTOR" 				default="1.0"/>
<arg name="vmax" 					        default="1.2"/>
<arg name="amax" 					        default="10"/>
<arg name="random_initial_x" 					default="0.0"/>
<arg name="random_goal_x" 					default="1.0"/>
<arg name="random_up_lim_y" 			                default="2"/>
<arg name="random_low_lim_y" 			                default="4"/>
<arg name="random_up_lim_y_initial" 			        default="-0.1"/>
<arg name="random_low_lim_y_initial" 			        default="0.1"/>
<arg name="random_up_lim_y_right" 			        default="-3.5"/>
<arg name="random_low_lim_y_left" 			        default="3.5"/>
<arg name="random_up_lim_x_up" 			                default="3.5"/>
<arg name="random_low_lim_x_down" 			        default="-3.5"/>
<arg name="alt" 			                        default="1.2"/>
<arg name="vel" 			                        default="3"/>
<arg name="acc" 			                        default="3.5"/>
<arg name="dynamic_env" 			                default="True"/>
<arg name="thresh_replan" 			                default="1.5"/>
<arg name="thresh_no_replan" 			                default="2.0"/>
<arg name="file_name" 					        default="/home/liang/workspace/ros/aerostack_catkin_ws/src/aerostack_stack/stack_devel/Depth_Based_RRT_STAR/launch/document.txt"/>



<node pkg="Depth_BASED_RRT_STAR" type="fast_replan_using_sdf_node" name="fast_replan_using_sdf_node" output="screen">
	<param name="configs_path" 	value="$(arg configs_path)" type="string"/>
	
	<param name = "Xmin"     					value="$(arg Xmin)" 	 			type="double" />
	<param name = "Xmax"     					value="$(arg Xmax)" 	 			type="double" />
	<param name = "Ymin"     					value="$(arg Ymin)" 	 			type="double" />
	<param name = "Ymax"     					value="$(arg Ymax)" 	 			type="double" />
	<param name = "END_X"     					value="$(arg END_X)" 	 			type="double" />
	<param name = "END_Y"     					value="$(arg END_Y)" 	 			type="double" />
        <param name = "END_Z"     					value="$(arg END_Z)" 	 			type="double" />
        <param name = "amax"     					value="$(arg amax)" 	 			type="double" />
        <param name = "vmax"     					value="$(arg vmax)" 	 			type="double" />
	<param name = "RRTSTAR_NEIGHBOR_FACTOR"     			value="$(arg RRTSTAR_NEIGHBOR_FACTOR)" 	 	type="double" />
        <param name = "random_initial_x"     			        value="$(arg random_initial_x)" 	 	type="double" />
        <param name = "random_goal_x"     				value="$(arg random_goal_x)" 	 		type="double" />
        <param name = "random_up_lim_y"     			        value="$(arg random_up_lim_y)" 	 		type="double" />
        <param name = "random_low_lim_y"     			        value="$(arg random_low_lim_y)" 	 	type="double" />
        <param name = "random_up_lim_y_initial"     		        value="$(arg random_up_lim_y_initial)" 	 	type="double" />
        <param name = "random_low_lim_y_initial"     		        value="$(arg random_low_lim_y_initial)" 	type="double" />
        <param name = "random_up_lim_y_right"     			value="$(arg random_up_lim_y_right)" 	 	type="double" />
        <param name = "random_low_lim_y_left"     			value="$(arg random_low_lim_y_left)" 	 	type="double" />
        <param name = "random_up_lim_x_up"     			        value="$(arg random_up_lim_x_up)" 	 	type="double" />
        <param name = "random_low_lim_x_down"     			value="$(arg random_low_lim_x_down)" 	 	type="double" />
        <param name = "alt"     		                        value="$(arg alt)" 	                        type="double" />
        <param name = "vel"     		                        value="$(arg vel)" 	                        type="double" />
        <param name = "acc"     		                        value="$(arg acc)" 	                        type="double" />
        <param name = "thresh_replan"     		                value="$(arg thresh_replan)" 	                type="double" />
        <param name = "thresh_no_replan"     		                value="$(arg thresh_no_replan)" 	        type="double" />
        <param name = "dynamic_env"     		                value="$(arg dynamic_env)" 	                type="bool" />
        <param name = "file_name"     				        value="$(arg file_name)" 	 		type="string" />

</node>

</launch>
